python_modules/libraries/dagster-components/dagster_components/utils.py

```diff
@@ -143,3 +155,77 @@ def load_module_from_path(module_name, path) -> ModuleType:
     assert spec.loader, "Must have a loader"
     spec.loader.exec_module(module)
     return module
+
+
+# ########################
+# ##### TOML MANIPULATION
+# ########################
+
+
+@contextmanager
+def modify_toml(path: Path) -> Iterator[tomlkit.TOMLDocument]:
+    with open(path) as f:
+        toml = tomlkit.parse(f.read())
+    yield toml
```

Comment:
the implication is that the caller will then modify this value? Is that better than using immutable objects and writing a deep copy of it?

Reply:
Personally I think modification is fine here since the object only springs into existence inside this function and is then discarded. Also it's unclear how we would use immutable objects here, `TOMLDocument` is a mutable object, and we need to use `TOMLDocument` instead of plain python objects in order to preserve comments/formatting in the TOML doc.